Observer/KlaviyoOauthObserver.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Message\ManagerInterface as MessageManager;
+use Magento\Integration\Model\IntegrationFactory;
+use Magento\Integration\Model\AuthorizationService;
+use Magento\Integration\Model\OauthService;
+
+
+class KlaviyoOauthObserver implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var \Klaviyo\Reclaim\Helper\ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * @var ManagerInterface $messageManager
+     */
+    protected $_messageManager;
+
+    /**
+     * @var IntegrationFactory $integrationFactory
+     */
+    protected $_integrationFactory;
+
+
+    /**
+     * @param ScopeSetting $klaviyoScopeSetting
+     * @param MessageManager $messageManager
+     * @param IntegrationFactory $integrationFactory
+     * @param AuthorizationService $authorizationService
+     * @param OauthService $oauthService
+     */
+    public function __construct(
+        ScopeSetting $klaviyoScopeSetting,
+        MessageManager $messageManager,
+        IntegrationFactory $integrationFactory,
+        AuthorizationService $authorizationService,
+        OauthService $oauthService
+    ) {
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+        $this->_messageManager = $messageManager;
+        $this->_integrationFactory = $integrationFactory;
+        $this->_authorizationService = $authorizationService;
+        $this->_oauthService = $oauthService;
+    }
+
+    /*
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        if (empty($this->_klaviyoScopeSetting->getPublicApiKey())) {
+         throw new \Magento\Framework\Exception\StateException(__('To set up Oauth for Klaviyo, first save a <strong>Public Klaviyo API Key</strong> on the "General" tab.'));
```

Comment:
this isn't the correct language, you don't save a public key, i'd say they should get the public api key -- is it worth adding in the hc link on how to find it:

https://help.klaviyo.com/hc/en-us/articles/115005062267-Manage-Your-Account-s-API-Keys#your-public-api-key-site-id2